cql3/statements/alter_type_statement.cc

```diff
@@ -87,6 +87,21 @@ void alter_type_statement::do_announce_migration(database& db, ::keyspace& ks, b
         throw exceptions::invalid_request_exception(format("No user type named {} exists.", _name.to_string()));
     }
 
+    for (auto&& schema : ks.metadata()->cf_meta_data() | boost::adaptors::map_values) {
```

Comment:
Why just the keyspace? I think a table in one keyspace can refer to types in another.

Reply:
In `cql3_type::raw_ut::prepare_internal(const sstring& keyspace, const user_types_metadata& user_types)` we have 
```
if (_name.has_keyspace()) {
    // The provided keyspace is the one of the current statement this is part of. If it's different from the keyspace of
    // the UTName, we reject since we want to limit user types to their own keyspace (see #6643)
    if (!keyspace.empty() && keyspace != _name.get_keyspace()) {
        throw exceptions::invalid_request_exception(sprint("Statement on keyspace %s cannot refer to a user type in keyspace %s; "
                                                           "user types can only be used in the keyspace they are defined in",
                                                        keyspace, _name.get_keyspace()));
    }
```
So if I understand correctly, the current implementation does not allow this.

Reply:
Ah, I remembered differently. But you are correct.